docs/DeveloperGuide.md

```diff
@@ -1,34 +1,383 @@
 # Developer Guide
 
+- [Design & implementation](#design--implementation)
+    - [Architecture](#architecture)  
+    - [Ui component](#ui-component)  
+    - [Logic component](#logic-component)  
+    - [Model component](#model-component)  
+    - [Storage component](#storage-component)
+- [Product scope](#product-scope)
+    - [Target user profile](#target-user-profile)
+    - [Value proposition](#value-proposition)
+- [User stories](#user-stories)
+- [Use cases](#use-cases)
+- [Non-functional requirements](#non-functional-requirements) 
+- [Glossary](#glossary)
+- [Instruction for manual testing](#instructions-for-manual-testing)
+
+
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+
+
+![Architecture](diagrams/Architecture.png)
+
+
+The Architecture Diagram above gives an overview of how different components interacts 
+with each other in the app. 
+
+As the app starts, the main program initialises the Ui, Logic and Storage components.
```

Comment:
Perhaps you want to illustrate the startup sequence using a sequence diagram, rather than using box and arrow diagrams.